x-pack/plugins/endpoint/server/routes/resolver/queries/lifecycle.ts

```diff
@@ -3,60 +3,63 @@
  * or more contributor license agreements. Licensed under the Elastic License;
  * you may not use this file except in compliance with the Elastic License.
  */
+import { SearchResponse } from 'elasticsearch';
 import { ResolverQuery } from './base';
 import { JsonObject } from '../../../../../../../src/plugins/kibana_utils/public';
+import { ResolverEvent } from '../../../../common/types';
 
-export class LifecycleQuery extends ResolverQuery {
-  protected legacyQuery(endpointID: string, uniquePIDs: string[], index: string): JsonObject {
+/**
+ * Builds a query for retrieving life cycle information about a node (start, stop, etc).
+ */
+export class LifecycleQuery extends ResolverQuery<ResolverEvent[]> {
+  protected legacyQuery(endpointID: string, uniquePIDs: string[]): JsonObject {
     return {
-      body: {
-        query: {
-          bool: {
-            filter: [
-              {
-                terms: { 'endgame.unique_pid': uniquePIDs },
-              },
-              {
-                term: { 'agent.id': endpointID },
-              },
-              {
-                term: { 'event.kind': 'event' },
-              },
-              {
-                term: { 'event.category': 'process' },
-              },
-            ],
-          },
+      query: {
+        bool: {
+          filter: [
+            {
+              terms: { 'endgame.unique_pid': uniquePIDs },
+            },
+            {
+              term: { 'agent.id': endpointID },
+            },
+            {
+              term: { 'event.kind': 'event' },
+            },
+            {
+              term: { 'event.category': 'process' },
+            },
+          ],
         },
-        size: 10000,
-        sort: [{ '@timestamp': 'asc' }],
       },
-      index,
+      size: 10000,
```

Comment:
There really should only be a couple lifecycle events for each node, there could be a case where we search for soooo many IDs at the same time at it causes the response to cap at 10k. That's unlikely after the `msearch` functionality goes it but it might be worth adding functionality here to limit the ids to like 1000 or something.

Reply:
:grey_question:  For the unlikely cases where we do hit this limit, it may be worth thinking about how we message that to a user.